src/main/java/org/mamute/infra/NormalizerBrutal.java

```diff
@@ -1,18 +1,34 @@
 package org.mamute.infra;
 
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
 import java.text.Normalizer;
 import java.text.Normalizer.Form;
 import java.util.Locale;
 import java.util.regex.Pattern;
 
 public class NormalizerBrutal {
-	public static String toSlug(String input) {
+	public static String toSlug(String input, boolean urlEncodeNonNormalizableCharacter)  {
 		if(input == null || input.isEmpty()) return "";
-		Pattern NONLATIN = Pattern.compile("[^\\w-]");
+
+		String nonLatinPattern = urlEncodeNonNormalizableCharacter ? "[^\\P{M}]" : "[^\\w-]";
```

Comment:
You are right that I didn't do the best job explaining what is happening.

So this is what we want:
- normalize characters like čćšž etc. (also called base-glyph, they match with P{M} in unicode)
- NOT normalize characters like  + or # or ! or . and rather URL encode them

If we would normalize character like #, it would first be decomposed to several characters and than URL encoded, which would result in something that is in fact not URL-encoded representation of a hash. 

In case we don't want to URL encode, then we SHOULD normalize all characters and strip out non ASCII representation of them at the end (thus \w)